console/views/submit.py

```diff
@@ -0,0 +1,194 @@
+
+    if os.path.exists(tmp_image_path):
+        os.unlink(tmp_image_path)
+
+    with ConsoleFtp() as ftp:
+        ftp._ftp.retrbinary("RETR " + PATHS['image'] + "/" + filename, open(tmp_image_path, 'wb').write)
+
+    return tmp_image_url
+
+
+def get_file_contents(datatype, filename):
+
+    with ConsoleFtp() as ftp:
+        return ftp.get_file_contents(datatype, filename)
+
+
+@submit_bp.route('/submit', methods=['POST', 'GET'])
+@flask_security.login_required
+def submit():
+    if request.method == 'POST':
+        data = request.get_data().decode('UTF8')
+
+        logger.debug(" [x] Encrypting data")
+
+        unencrypted_json = json.loads(data)
+
+        no_of_submissions = int(unencrypted_json['quantity'])
+
+        with open("./keys.yml") as file:
+            secrets_from_file = yaml.safe_load(file)
+
+        key_store = KeyStore(secrets_from_file)
+
+        tx_id = unencrypted_json['survey']['tx_id']
+
+        for _ in range(0, no_of_submissions):
+            # If submitting more than one then randomise the tx_id
+            if tx_id is not None:
+                tx_id = str(uuid.uuid4())
+                unencrypted_json['survey']['tx_id'] = tx_id
+                logger.info("Auto setting tx_id", tx_id=tx_id)
+
+            payload = encrypt(unencrypted_json['survey'], key_store, 'submission')
+            send_payload(payload, tx_id, 1)  # let the loop handle the submission
+
+        return data
+    else:
+        return render_template('submit.html', enable_empty_ftp=settings.ENABLE_EMPTY_FTP)
+
+
+def client_error(error=None):
+    logger.error(error)
+    message = {
+        'status': 400,
+        'message': error,
+        'uri': request.url
+    }
+    resp = jsonify(message)
+    resp.status_code = 400
+
+    return resp
+
+
+def get_paginate_info(ru_ref):
+    # This is a little hacky as pagination.start or
+    # pagination.end does not work in the view.
+    info = "Found <b>{total}</b>,"
+    if ru_ref:
+        info += " matching ru_ref: <b>{0}</b>,".format(ru_ref)
+    info += " displaying <b>{start} - {end}</b>"
+    return info
+
+
+@submit_bp.route('/validate', methods=['POST', 'GET'])
+def validate():
+    if request.method == 'POST':
+        payload = request.get_data()
+
+        logger.debug("Validating json...")
+
+        logger.debug("Validate URL: {}".format(settings.SDX_VALIDATE_URL))
+
+        r = requests.post(settings.SDX_VALIDATE_URL, data=payload)
+
+        return jsonify(json.loads(r.text))
+    else:
+        logger.info('Failed validation')
+        ftp_data = get_ftp_contents()
+
+        return render_template('submit.html', ftp_data=json.dumps(ftp_data))
+
+
+@submit_bp.route('/ftp.json')
+def ftp_list():
+    return jsonify(get_ftp_contents())
+
+
+@submit_bp.route('/view/<datatype>/<filename>')
+def view_file(datatype, filename):
+    if filename.lower().endswith(('jpg', 'png')):
+        return '<img style="width:100%;" src="/' + get_image(filename) + '" />'
```

Comment:
This should be a template if its needed

Reply:
I've removed this for now as it wasn't working anyway, will be fixed as part of the receiver ticket.